src/Sylius/Bundle/ApiBundle/DataPersister/AddressBookDataPersister.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\DataPersister;
+
+use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;
+use Sylius\Bundle\ApiBundle\Provider\CustomerProvider;
+use Sylius\Component\Core\Model\AddressInterface;
+use Sylius\Component\User\Model\UserInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Exception\TokenNotFoundException;
+
+final class AddressBookDataPersister implements ContextAwareDataPersisterInterface
+{
+    /** @var ContextAwareDataPersisterInterface */
+    private $decoratedDataPersister;
+
+    /** @var TokenStorageInterface */
+    private $tokenStorage;
+
+    /** @var CustomerProvider */
+    private $customerProvider;
+
+    public function __construct(
+        ContextAwareDataPersisterInterface $decoratedDataPersister,
+        TokenStorageInterface $tokenStorage,
+        CustomerProvider $customerProvider
+    ) {
+        $this->decoratedDataPersister = $decoratedDataPersister;
+        $this->tokenStorage = $tokenStorage;
+        $this->customerProvider = $customerProvider;
+    }
+
+    public function supports($data, array $context = []): bool
+    {
+        return $data instanceof AddressInterface;
+    }
+
+    public function persist($data, array $context = [])
+    {
+        $token = $this->tokenStorage->getToken();
+        if ($token === null) {
+            throw new TokenNotFoundException();
+        }
+
+        /** @var UserInterface $loggedUser */
+        $loggedUser = $token->getUser();
+
+        if ($loggedUser instanceof UserInterface) {
+            $customer = $this->customerProvider->provide($loggedUser->getEmail());
+
+            $data->setCustomer($customer);
+            $data->setFirstName($customer->getFirstName());
+            $data->setLastName($customer->getLastName());
```

Comment:
Do we have feature like this? That if you fill full name in address it will be replaced by customer account full name anyway? It seems like a bug rather then a feature.
```suggestion
```